cmd/batch-handlers.go

```diff
@@ -526,7 +526,7 @@ func toObjectInfo(bucket, object string, objInfo miniogo.ObjectInfo) ObjectInfo
 
 	_, ok = oi.UserDefined[xhttp.AmzStorageClass]
 	if !ok {
-		oi.UserDefined[xhttp.AmzStorageClass] = objInfo.StorageClass
+		oi.UserDefined[xhttp.AmzStorageClass] = globalAPIConfig.storageClass(objInfo.StorageClass)
```

Comment:
batch-jobs interface is a MinIO interface and unencumbered by external clients. So, in this case we can return the unmodified storage class.